fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1139,7 +1149,26 @@ def _register_pre_backward_hooks(self, outputs: Any) -> Any:
         if not torch.is_grad_enabled():
             return outputs  # don't register hooks if grad isn't enabled
 
+        if self._is_root:
+            # This actually means that only root instance has
+            # _post_backward_callback_queued defined. Accidentally accessing this field
+            # will assert on all other instances, giving us a nice bug checker.
+            self._post_backward_callback_queued = False
+            # Make sure post backward callback was fired in the preivous iteration
+            assert self._post_backward_callback_fired
```

Comment:
nice check!

Reply:
actually, I am wondering if you can assert this using TrainingState? Just check the state to be BACKWARD_POST should be the same?